ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitKudf.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.string;
+
+import com.google.common.base.Splitter;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import java.util.Arrays;
+import java.util.List;
+
+@UdfDescription(name = SplitKudf.NAME, author = "Confluent",
+    description = "Splits a string into an array of substrings based on a delimiter.")
+public class SplitKudf {
+  protected static final String NAME = "split";
+
+  private static final String EMPTY_DELIMITER = "";
+
+  @Udf(description = "Splits a string into an array of substrings based on a delimiter.")
+  public List<String> split(
+      @UdfParameter(value = "string",
+          description = "The string to be split. If NULL, then function returns NULL.")
+      final String string,
+      @UdfParameter(value = "delimiter",
+          description = "The delimiter to split a string by. If NULL, then function returns NULL.")
+      final String delimiter) {
+    if (string == null || delimiter == null) {
+      return null;
+    }
+
+    // Java split() accepts regular expressions as a delimiter, but the behavior of this UDF split()
+    // is to accept only literal strings. This method uses Guava Splitter instead, which does not
+    // accept any regex pattern. This is to avoid a confusion to users when splitting by regex
+    // special characters, such as '.' and '|'.
+
+    try {
+      // Guava Splitter does not accept empty delimiters. Use the Java split() method instead.
+      if (delimiter.isEmpty()) {
+        return Arrays.asList(string.split(EMPTY_DELIMITER));
```

Comment:
as a small performance optimization we can pre-compile the Pattern for empty delimiter

Reply:
The Java string.split() method accepts a String as a parameter only (no Pattern).

Reply:
If you look at the implementation of string.split(), it uses a Pattern underneath (though it's a pretty straightforward Pattern :P)

Reply:
I cannot make it work, I tried the "." character to split any character, but I get empty strings in the array. See the different examples:

```
String str = "abc";
    
System.out.println(Splitter.on(CharMatcher.any()).splitToList(str));
System.out.println(Splitter.onPattern(".").splitToList(str));
System.out.println(Splitter.on(".").splitToList(str));
System.out.println(Arrays.asList(str.split("")));
System.out.println(Arrays.asList(str.split(".")));
```

```
[, , , ]
[, , , ]
[abc]
[a, b, c]
[]
```

Reply:
Again, not a big deal so feel free to ignore, but what I meant was this:
```
Pattern EMPTY_DELIMITER = Pattern.compile("");
...
EMPTY_DELIMITER.split("input");
```
which is what `String#split` does underneath.

Reply:
yep - for performance it would be better to compile the `Pattern` only once rather than on every invocation. Not a biggie though

Reply:
Aaa, thanks. That worked. I just added.